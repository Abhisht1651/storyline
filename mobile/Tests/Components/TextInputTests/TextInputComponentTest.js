import React from 'react'
import { shallow } from 'enzyme'
import { TextInput } from 'react-native'
import TextInputComponent from '../../../App/Components/TextInput/TextInputComponent'

describe('TextInputComponent', () => {
  it('TextInputComponent  renders correctly', () => {
    const wrapper = shallow(<TextInputComponent />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('TextInputComponent ', () => {
  it('should have prop onChangeText', () => {
    const wrapper = shallow(<TextInputComponent />)
    expect(wrapper.find(TextInput).prop('onChangeText')).toBeTruthy()
  })

  it('should change value after entering text', () => {
    const wrapper = shallow(<TextInputComponent />)
    wrapper.find(TextInput).prop('onChangeText')('Random')
    expect(wrapper.find(TextInput).prop('value')).toEqual('Random')
  })
})
