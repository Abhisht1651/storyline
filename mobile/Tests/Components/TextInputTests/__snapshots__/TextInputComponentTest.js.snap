// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputComponent TextInputComponent  renders correctly 1`] = `
<Component
  allowFontScaling={true}
  autoCorrect={false}
  keyboardType="default"
  onChangeText={[Function]}
  placeholder="Full name"
  placeholderTextColor="#8F8FAF"
  rejectResponderTermination={true}
  selectionColor="#7B46E4"
  spellCheck={false}
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Medium",
        "fontSize": 30,
        "height": 49.285714285714285,
        "letterSpacing": -0.12,
        "paddingLeft": 0,
        "paddingVertical": 0,
        "textAlign": "left",
        "width": "100%",
      },
      Object {
        "borderBottomColor": "#8F8FAF",
      },
    ]
  }
  underlineColorAndroid="transparent"
  value=""
/>
`;
